### PR TITLE
Minimizing bob.jar size

### PR DESCRIPTION
**Pull request by JCash**
In an effort to keep our toolchain as small as possible, we've removed some old and redundant packages.

## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java [hunk 1]
```diff
@@ -147,38 +147,11 @@ public static void initLua() {
...
-            if (classes_dex != null) {
-                File f = new File(rootFolder, "lib/classes.dex");
-                atomicCopy(classes_dex, f, false);
-            }
-
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
+    public static File getRootFolder() {
+        return rootFolder;
```
Comment (JCash) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java:151]: Better to move the platform specific code to the actual file that uses it

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidBundler.java [hunk 2]
```diff
@@ -110,6 +114,81 @@ private static String getJavaBinFile(String file) {
...
     }
 
+    private static void extractFile(File zipFile, String fileName, File outputFile) throws IOException {
+        // Wrap the file system in a try-with-resources statement
+        // to auto-close it when finished and prevent a memory leak
+        try (FileSystem fileSystem = FileSystems.newFileSystem(zipFile.toPath(), null)) {
+            Path fileToExtract = fileSystem.getPath(fileName);
+            Files.copy(fileToExtract, outputFile.toPath());
+        }
+    }
```
Comment (JCash) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidBundler.java:124]: How to extract a file from a zip file (e.g. a .jar file)

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidBundler.java [hunk 3]
```diff
@@ -110,6 +114,81 @@ private static String getJavaBinFile(String file) {
...
+
+    private static String getAapt2Name()
+    {
+        if (Platform.getHostPlatform() == Platform.X86_64Win32)
+            return "aapt2.exe";
+        return "aapt2";
+    }
+
+    private static void initAndroid() {
```
Comment (JCash) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidBundler.java:133]: Moved from Bob.java

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidBundler.java [hunk 4]
```diff
@@ -110,6 +114,81 @@ private static String getJavaBinFile(String file) {
...
+                {
+                    platformName = "linux";
+                }
+
+                File aapt2 = new File(bundletool.getParent(), getAapt2Name());
+                if (!aapt2.exists())
+                {
+                    extractFile(bundletool, platformName + "/" + getAapt2Name(), aapt2);
+                    aapt2.setExecutable(true);
+                }
```
Comment (JCash) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidBundler.java:183]: If it doesn't exist, extract it

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaBuilder.java [hunk 5]
```diff
@@ -237,6 +237,7 @@ private String getChunkName(Task<Void> task) {
         return chunkName;
     }
 
+    /* We currently prefer source code over plain lua byte code due to the smaller size
```
Comment (JCash) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaBuilder.java:240]: Commented out unused luac related code

File: TOOLS_LICENSES.md [hunk 6]
```diff
@@ -21,4 +21,3 @@ The following software is used by the Defold editor and command line tools:
     * [IconExe](/licenses/NOTICE-iconexe) - Used for replacing icons bundled in the .exe from bob.jar
   * Apache 2.0
     * [bundletool-all.jar](/licenses/NOTICE-bundletool) - Used by the command line tools when creating an Android App Bundle or APK.
-    * [aapt2](/licenses/NOTICE-aapt2) - Used by the command line tools to compile and package Android resources.
```
Comment (JCash) [inline TOOLS_LICENSES.md:24]: We are now only using the `bundletool-all.jar`

**Review by britzl — APPROVED** [review]

**Pull request merged**